### PR TITLE
Modified can parser to run it from PlotJuggler

### PR DESCRIPTION
**Pull request by jwooning**
- PlotJuggler uses dynamic structs, so added optional dependency to capnp struct files.
- Added CANParser which parses all messages and signals

### CODE REVIEW

File: can/parser.cc [hunk 1]
```diff
@@ -225,6 +212,65 @@ void CANParser::update_string(const std::string &data, bool sendcan) {
...
+    uint8_t dat[8] = {0};
+    memcpy(dat, cmsg.getDat().begin(), cmsg.getDat().size());
+
+    state_it->second.parse(sec, cmsg.getBusTime(), dat);
+  }
+}
+#endif
+
+void CANParser::UpdateCans(uint64_t sec, const capnp::DynamicStruct::Reader& cmsg) {
+  // assume message struct is `cereal::CanData` and parse
```
Comment (pd0wm) [inline can/parser.cc:243]: can we add an assert for this?

File: can/dbc.cc [hunk 2]
```diff
@@ -2,15 +2,11 @@
 
 #include "common_dbc.h"
 
-namespace {
```
Comment (adeebshihadeh) [inline can/dbc.cc:5]: why remove the namespace?

File: can/parser.cc [hunk 3]
```diff
@@ -34,45 +33,45 @@ bool MessageState::parse(uint64_t sec, uint16_t ts_, uint8_t * dat) {
 
     DEBUG("parse 0x%X %s -> %lld\n", address, sig.name, tmp);
 
-    if (sig.type == SignalType::HONDA_CHECKSUM) {
+    if (!ignore_checksum && sig.type == SignalType::HONDA_CHECKSUM) {
```
Comment (adeebshihadeh) [inline can/parser.cc:36]: can we just make two outer if's with ignore_checksum and ignore_counter?

File: can/parser.cc [hunk 4]
```diff
@@ -169,45 +168,34 @@ CANParser::CANParser(int abus, const std::string& dbc_name,
...
-        continue;
-      }
+  dbc = dbc_lookup(dbc_name);
+  assert(dbc);
+  init_crc_lookup_tables();
 
-      if (cmsg.getDat().size() > 8) continue; //shouldn't ever happen
-      uint8_t dat[8] = {0};
-      memcpy(dat, cmsg.getDat().begin(), cmsg.getDat().size());
+  for (int i=0; i<dbc->num_msgs; i++) {
```
Comment (adeebshihadeh) [inline can/parser.cc:179]: whitespace

File: can/parser.cc [hunk 5]
```diff
@@ -169,45 +168,34 @@ CANParser::CANParser(int abus, const std::string& dbc_name,
...
+    const Msg* msg = &dbc->msgs[i];
+    MessageState state = {
+      .address = msg->address,
+      .size = msg->size,
+      .ignore_checksum = ignore_checksum,
+      .ignore_counter = ignore_counter,
+    };
 
-      state_it->second.parse(sec, cmsg.getBusTime(), dat);
+    for (int j=0; j<msg->num_sigs; j++) {
```
Comment (adeebshihadeh) [inline can/parser.cc:188]: whitespace

**Pull request merged**